### ISSUE TITLE
QuickHash doesn't contain correct version information on macOS

### ISSUE THREAD

**Issue by core-code** [tedsmith/quickhash#44]
<key>CFBundleShortVersionString</key>
  <string>0.1</string>
  <key>CFBundleVersion</key>
  <string>1</string>

these are wrong and should contain "3.0.2"

**Comment by core-code** [comment]
thanks looking forward to this being fixed in 3.0.3

**Comment by tedsmith** [comment]
V3.0.3 was released last night. Are you able to see/check if whatever command you used to check last time reports correctly now? I've ensured that all the fields where a version number can be input into the project IDE was populated with 3.0.3, so if it is not right still, then I do not know how to fix!

**Comment by core-code** [comment]
sorry to bring bad news but i've downloaded version v3.0.3 and it still has version "0.1" when selected in the Finder

if you have a Mac, its easy to reproduce the problem by selecting the app in the Finder.
if you don't have a Mac, it should be possible to read the Info.plist files with a few lines of Python, i could help you there.

i can also have a look at your sources to see if i can fix the problem. which commit contains the changes intended to fix that so far?

**Comment by core-code** [comment]
python (2.x?) code to check macOS version numbers from linux/windows:


```
>>> import plistlib
>>> pl = plistlib.readPlist("path/to/QuickHash.app/Contents/Info.plist")
>>> print pl['CFBundleShortVersionString']
>>> print pl['CFBundleVersion']

```

**Comment by core-code** [comment]
i've had a short look at it but didn't get far, didn't touch pascal since the 90s ;) 
i think the best bet is to ask here:
https://forum.lazarus.freepascal.org/index.php/board,33.0.html
how to get the CFBundleVersion & CFBundleShortVersionString set correctly.

if Lazarus can't do it, i guess its easy to just post-process the Info.plist yourself, but i am not sure if that would interfere with your code-signing setup

**Comment by tedsmith** [comment]
@core-code forgive me for returning to this 2 years late! 

I am aiming to have this set properly for the release of v3.3.0 on this coming Saturday 29th. I am aware that still the app generation is not set properly from the project version number but, as you describe, I (as the dev) can edit the info.plist file and set it manually. I tried it last night as part of a test build and it seemed to work OK. 

I have suggested this be set automatically within Lazarus for future versions (there is a way [to do it programmatically](https://wiki.lazarus.freepascal.org/Mac_Show_Application_Title,_Version,_and_Company) but that seems a bit unnecessary to me given the info needed is set already in the project version data), because it is a nuisance to have to remember to set it manually I fancy. Nevertheless, thanks for raising it and hopefully v3.3.0 will be set properly. 

I note also your link to that project. I stopped doing code signed distribution because of the cost of the certs. But I now have my eye on SigStore (https://github.com/sigstore/sigstore) awaiting the availability of that. Bit of a game changer for developers that will be, I think.
